pkg/common/resource.go

```diff
@@ -87,6 +86,48 @@ func GetPodResource(pod *v1.Pod) (resource *si.Resource) {
 	return podResource
 }
 
+func containerResource(pod *v1.Pod, i int) (resource *si.Resource) {
+	// K8s pod InPlacePodVerticalScaling from:
+	// alpha: v1.27
+	// beta: v1.31?
+	// If AllocatedResources are present, these need to be used in preference to pod resource requests.
+	// Additionally, if the Resize pod status is InProgress, then the maximum of the request and allocated values need
+	// to be used.
+	requested := pod.Spec.Containers[i].Resources.Requests
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return getResource(requested)
+	}
+	allocated := pod.Status.ContainerStatuses[i].AllocatedResources
```

Comment:
Do `ContainerStatuses` and `Containers`  have same order?

Reply:
Yes, this is part of the spec.

Reply:
See https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md#scheduler-and-api-server-interaction for more details.